docs/orchestration/faq/faq.md

```diff
@@ -0,0 +1,228 @@
+
+And if you have a strict requirement that every process must run in a container, consider using the [KubernetesAgent](orchestration/agents/kubernetes.html#requirements) instead.
+
+### How can I change the number of DaskExecutor workers based on a Parameter value?
+
+You can pass a Callable to the DaskExecutor that sizes the DaskExecutor dynamically at runtime.
+
+```python
+from prefect import Flow
+from prefect.executors import DaskExecutor
+
+def dynamic_executor():
+    from distributed import LocalCluster
+    # could be instead some other class e.g. from dask_cloudprovider.aws import FargateCluster
+    return LocalCluster(n_workers=prefect.context.parameters["n_workers"])
+
+with Flow("example", executor=DaskExecutor(cluster_class=dynamic_executor)) as flow:
+    flow.add_task(Parameter("n_workers", default=5))
+```
+
+### How can I force a task to Fail based on a given condition?
+
+If the task itself throws an error, the task will also fail as a result. If there is some other condition that you want to trigger a Failed state, the most direct way is by simply raising a FAIL signal:
+
+```python
+from prefect import task
+from prefect.engine.signals import FAIL
+
+@task
+def my_task(condition: bool=True):
+  if condition:
+	  raise FAIL('this task failed')
+	else:
+		return "all is good"
+```
+
+### I can't connect to the Prefect Server API from other machines. What am I missing?
+
+First, make sure that port 4200 is opened for outside connections to access. Also check that the API is working on the Server deployment by going to the Interactive API and testing that the API is healthy. The `hello` endpoint can be used as follows:
+
+```graphql
+query {
+  hello
+}
+```
+
+If it is working, you might need to start the Server with the expose flag as follows: `prefect server start --expose`. This will allow outside connections. You can find more information in this [Github issue](https://github.com/PrefectHQ/prefect/issues/4963).
+
+### Why does Prefect mark Flow runs with `No heartbeat detected from the remote task; marking the run as failed`?
+
+Prefect Flows have heartbeats that signal to Prefect Cloud that your Flow is alive. If Prefect didn’t have heartbeats, Flows that lose communication and die would permanently be shown with a Running state in the UI. Most of the time, we have seen “no heartbeat detected” to be a sign of memory issues. In version 0.15.4 and above, additional logging has been added that propagates the real error in the event this error happens. 
+
+You can [configure heartbeats to use threads instead of processes.](orchestration/concepts/services.html#heartbeat-configuration) This has proven to be more stable for a lot of users. In the event that the Flow continues to fail after switching to using threads, some users have had success in making the failing task a subflow, and then turning off heartbeats for the subflow.
```

Comment:
```suggestion
You can [configure heartbeats to use threads instead of processes.](/orchestration/concepts/services.html#heartbeat-configuration) This has proven to be more stable for many users. In the event that the flow continues to fail after switching to using threads, some users have had success in making the failing task a subflow, and then turning off heartbeats for the subflow.
```